File: Date System with Percentages/C#/Penanggalan.cs
using System;

public class Time {
    int Leap;
    int Tahun;
    int Bulan;
    int Tanggal;
    int Harke;
    int Harmun;
    double Jam;
    double Menit;
    double Detik;
    double Persen;
    double Permun;
    double DateCal;

    public void DisplayDateAndTime() {
        Console.WriteLine($"Tahun                    : {Tahun}");
        Console.WriteLine($"Bulan                    : {Bulan} ("+DateTime.Now.ToString("MMMM")+")");
        Console.WriteLine($"Tanggal                  : {Tanggal}");
        Console.WriteLine($"Hari                     : "+DateTime.Now.ToString("dddd"));
        Console.WriteLine($"Hari ke (dalam Setahun)  : {Harke}");
        Console.WriteLine($"Sisa Hari (Tahun depan)  : {Harmun}");
        Console.WriteLine($"Persentase dalam Setahun : "+Math.Round(Persen, 6)+"%");
        Console.WriteLine($"Persentase Hitung Mundur : "+Math.Round(Permun, 6)+"%");
        Console.WriteLine($"Jam                      : {Jam}");
        Console.WriteLine($"Menit                    : {Menit}");
        Console.WriteLine($"Detik                    : {Detik}");
    }

    public void DisplayCurrentTime() {
        Console.WriteLine(DateTime.Now.ToString("dddd, dd MMMM yyyy"));
        Console.WriteLine(DateTime.Now.ToString("HH:mm:ss"));
        System.Console.WriteLine("{0}:{1}:{2}", Jam, Menit, Detik);
        Console.WriteLine($"Day-{Harke} ("+Math.Round(Persen, 4)+"%)");
    }

    public Time(System.DateTime dt) {
        Tahun = dt.Year;
        Bulan = dt.Month;
        Tanggal = dt.Day;
        Jam = dt.Hour;
        Menit = dt.Minute;
        Detik = dt.Second;
        
        if (Tahun % 400 == 0){
          Leap = 366;
        }
        else if (Tahun % 100 == 0){
          Leap = 365;
        }
        else if (Tahun % 4 == 0){
          Leap = 366;
        }
        else {
          Leap = 365;
        }
        
        Harke = int.Parse(dt.DayOfYear.ToString());
        Harmun = 1+Leap-Harke;
        
        DateCal = (Harke-1)+(Jam/24)+(Menit/1440)+(Detik/86400);
        Persen = (double)(100*DateCal/Leap);
        Permun = 100-Persen;
    }
}

public class Penanggalan {
    static void Main(string[] args) {
        System.DateTime currentTime = System.DateTime.Now;
        Time waktu = new Time(currentTime);
        waktu.DisplayDateAndTime();
        Console.WriteLine("");
        waktu.DisplayCurrentTime();
    }
}

File: Date System with Percentages/Java/Penanggalan.java
import java.util.*;
import java.util.Date;
import java.text.*;
import java.math.RoundingMode;

public class Penanggalan {
    int Leap, Tahun, Bulan, Tanggal;
    int Harmun, Harke;
    float Jam, Menit, Detik;
    float Persen, Permun, DateCal;

    public void DateDisplay() {
        Date month = new Date();
        SimpleDateFormat mn = new SimpleDateFormat("MMMMM");
        Date day = new Date();
        SimpleDateFormat dy = new SimpleDateFormat("EEEEE");

        Calendar cal = Calendar.getInstance();

        Tahun = cal.get(Calendar.YEAR);
        Bulan = cal.get(Calendar.MONTH)+1;
        Tanggal = cal.get(Calendar.DATE);

        System.out.println("Tahun                    : " + Tahun);
        System.out.println("Bulan                    : " + Bulan+" ("+mn.format(month)+")");
        System.out.println("Tanggal                  : " + Tanggal);
        System.out.println("Hari                     : " + dy.format(day));
    }

    public void DisplayCurrentTime(){
        String shortPattern = "dd-MM-yyyy";
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(shortPattern);
        String shortDate = simpleDateFormat.format(new Date());
        
        String longPattern = "EEEEE, dd MMMMM yyyy";
        SimpleDateFormat dateFormat = new SimpleDateFormat(longPattern);
        String longDate = dateFormat.format(new Date());

        String pattern = "HH:mm:ss";
        SimpleDateFormat timeFormat = new SimpleDateFormat(pattern);
        String time = timeFormat.format(new Date());

        String dayYear = "D";
        SimpleDateFormat dayinayear = new SimpleDateFormat(dayYear);
        String harke = dayinayear.format(new Date());
        
        if (Tahun % 400 == 0){
          Leap = 366;
        }
        else if (Tahun % 100 == 0){
          Leap = 365;
        }
        else if (Tahun % 4 == 0){
          Leap = 366;
        }
        else {
          Leap = 365;
        }

        Harke = Integer.parseInt(harke);
        Harmun = 1+Leap-Harke;
        
        DateCal = (Harke-1)+((float)Jam/24)+((float)Menit/1440)+((float)Detik/86400);
        Persen = (float)(100*DateCal/Leap);
        Permun = 100-Persen;
        
        DecimalFormat df = new DecimalFormat("#.####");
        
        System.out.println("Hari ke (dalam Setahun)  : " + Harke);
        System.out.println("Sisa Hari (Tahun depan)  : " + Harmun);
        System.out.println("Persentase dalam Setahun : " + Persen+"%");
        System.out.println("Persentase Hitung Mundur : " + Permun+"%");
        System.out.println();
        
        System.out.println(shortDate);
        System.out.println(longDate);
        System.out.println(time);
        System.out.print("Day-"+harke);
        System.out.print(" ("+df.format(Persen)+"%)");
    }

    public void TimeDisplay() {
        Calendar tme = Calendar.getInstance();

        DecimalFormat wkt = new DecimalFormat("#");

        Jam = tme.get(Calendar.HOUR_OF_DAY);
        Menit = tme.get(Calendar.MINUTE);
        Detik = tme.get(Calendar.SECOND);

        System.out.println("Jam                      : " + wkt.format(Jam));
        System.out.println("Menit                    : " + wkt.format(Menit));
        System.out.println("Detik                    : " + wkt.format(Detik));
    }

    public static void main(String args[]) {
        Penanggalan tanggalWaktu = new Penanggalan();
        tanggalWaktu.DateDisplay();
        tanggalWaktu.TimeDisplay();
        tanggalWaktu.DisplayCurrentTime();
    }
}